ncigrafana/UsageDataset.py

```diff
@@ -664,4 +664,4 @@ def addprojectmembership(self, project, system, date, members_list, members_coun
                     date=date.date(),
                     members=members_csv,
                     members_count=members_count)
-        return self.db['ProjectMembership'].upsert(data, ['project_id', 'system_id', 'date', 'members', 'members_list'])
+        return self.db['ProjectMembership'].upsert(data, ['project_id', 'system_id', 'date'])
```

Comment:
I'm a bit confused, when you say:

> we have too many fields defined in the keys kwarg

aren't the "members" and "members_list" required bits of info to add to the db?

Reply:
I think it is only the database keys that are required. The other data is what is in(up)serted.

Reply:
@CodeGat Yes, as Aidan says all the fields are in the data object, the keys is to do with the upsert. Postgres needs to find a matching unique constraint and these are the fields that it uses to find it. It's basically a try/exception, so the fields specify which 'exception' you want to catch.  

In this example data would be `new_name` and `size`, keys would be `id`

```insert into dummy(id, name, size) values(1, 'new_name', 3) on conflict(id) do update set name = 'new_name', size = 3;  ```